Makefile

```diff
@@ -1,8 +1,19 @@
-CC ?= clang
 CFLAGS = -O3 -Ofast -Wno-unused-result
 LDFLAGS =
 LDLIBS = -lm
-INCLUDES =
+INCLUDES = 
+
+# AMD HIP, Nvidia and AMD support
+USE_HIP = false
```

Comment:
Consider the ?= operator.. Then I would be able to build like this:
USE_HIP=true make

also, for the CC , use ?= so that I can build with e.g. gcc if I don't have clang installed

Reply:
Good idea! I updated the makefile according to your suggestions.

Reply:
Nice, but iemove it for the include, it is only relevant for setting logics and compiler in my opinin :)

Reply:
Nice. But let INCLUDES have the += still. Also, consider rebasing this PR so it is only one big commit. That probably makes it feel less cluttered for Andrej. Also, his silence probably suggests that he needs time to review your changes. I for myself think it looks very interesting but I havent tested it.